.github/workflows/check_submodule.yml

```
@@ -0,0 +1,43 @@
+name: Validate Submodule Commits
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  submodule-commit-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Validate Submodule Commits
+        run: |
+          # Define latest release commits
```

```suggestion
          # Submodules are very easy to accidentally overwrite with old versions.
          # To make submodule updates more explicit and to prevent accidental changes we check the revisions explicitly here.
```

Maybe we can add a bit of context. Otherwise it looks quite pointless to do this check